temporalio/lib/temporalio/internal/worker/activity_worker.rb

```diff
@@ -236,8 +236,11 @@ def execute_activity(task_token, defn, start)
           remove_running_activity(task_token)
         end
 
-        def run_activity(activity, input)
+        def run_activity(defn, activity, input)
           result = begin
+            # Create the instance. We choose to do this before interceptors so that it is available in the interceptor.
+            activity.instance = defn.instance.is_a?(Proc) ? defn.instance.call : defn.instance # steep:ignore
```

Comment:
Could've done this by ensuring the first interceptor is always an interceptor that makes the instance

:wink:

Reply:
Not following this exactly. We originally had our interceptor make the instance, but that didn't provide the instance early enough for user interceptors. We don't have control over interceptor order to ensure any one is the first.